src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -79,6 +82,49 @@ export class HttpResourceFetcher implements ResourceFetcher {
         return this.downloadRequest()
     }
 
+    /**
+     * Requests for new content but additionally uses the given E-Tag.
+     * If no E-Tag is given it behaves as a normal request.
+     *
+     * @param eTag
+     * @returns object with optional content. If content is undefined it implies the provided
+     *          E-Tag matched the server's, so no content was in response. If the return value
+     *          is undefined then there was an error.
+     */
+    public async getNewETagContent(eTag?: string): Promise<{ content?: string; eTag: string } | undefined> {
+        try {
+            const response = await this.getResponseFromGetRequest(this.params.timeout, { eTag, gZip: true })
+
+            const eTagResponse = response.headers.etag
+            if (!eTagResponse) {
+                throw new Error(
+                    `This URL does not support E-Tags. Cannot use this function for: ${this.url.toString()}`
+                )
+            }
+
+            let contents: string | undefined = response.body.toString()
+            if (response.statusCode === 304) {
+                // Explanation: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match
+                contents = undefined
+                this.logger.verbose(`E-Tag, ${eTagResponse}, matched. No content downloaded from: ${this.url}`)
+            } else {
+                this.logger.verbose(`No E-Tag match. Downloaded content from: ${this.logText()}`)
+                if (this.params.onSuccess) {
+                    this.params.onSuccess(contents)
+                }
+            }
+
+            return { content: contents, eTag: eTagResponse }
+        } catch (err) {
+            const error = err as CancelError | RequestError
+            this.logger.verbose(
+                `Error downloading ${this.logText()}: %s`,
+                error.message ?? error.code ?? error.response?.statusMessage ?? error.response?.statusCode
+            )
+            return undefined
```

Comment:
Use `ToolkitError.chain` instead of swallowing. Let the caller decide if the exception needs to be handled.